Fix #37756: Correct formatting of hexadecimal imaginary parts

tsbranquinho opened the pull request:
The 'show' method for 'Complex' was not properly handling cases where the imaginary part was represented in hexadecimal ('0x'). This could lead to missing multiplication symbols '*' in the output, making the formatting inconsistent.

To fix this, a check using 'occursin("0x", repr(i))' was added to detect these cases and ensure proper formatting.

This improves the consistency of complex number representations and prevents ambiguities in the printed output.

tsbranquinho committed:
Fix #37756: Correct formatting of hexadecimal imaginary parts

The 'show' method for 'Complex' was not properly handling cases where
the imaginary part was represented in hexadecimal ('0x'). This could
lead to missing multiplication symbols '*' in the output, making the
formatting inconsistent.

To fix this, a check using 'occursin("0x", repr(i))' was added to
detect these cases and ensure proper formatting.

This improves the consistency of complex number representations and
prevents ambiguities in the printed output.

LilithHafner commented on base/complex.jl:
```diff
@@ -206,7 +206,7 @@ function show(io::IO, z::Complex)
         show(io, i)
     end
-    if !(isa(i,Integer) && !isa(i,Bool) || isa(i,AbstractFloat) && isfinite(i))
+    if !(isa(i,Integer) && !isa(i,Bool) || isa(i,AbstractFloat) && isfinite(i)) || occursin("0x",repr(i))
```
This is a good rule but it is better to implement it with a type check.
```suggestion
    if !(isa(i,Signed) || isa(i,AbstractFloat) && isfinite(i))
```

LilithHafner reviewed (commented):
Thanks for working on this! You have a good solution and should also add tests to make sure it works as expected.

tsbranquinho committed:
Refactor base/complex.jl: Improve readability with type check

Replaced explicit conditionals with a type check to make the code easier 
to understand and maintain. This ensures that the logic is clearer while 
preserving the original behavior.

Co-authored-by: Lilith Orion Hafner <lilithhafner@gmail.com>

tsbranquinho committed:
test: add tests for `show` method with various number formats

These additions improve test coverage and ensure robust handling of
different numeric representations in the `show` function.

This extends the test suite in reference to issue #37756.

LilithHafner commented on test/complex.jl:
```diff
@@ -1,6 +1,7 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 using LinearAlgebra
+using Test
```
```suggestion
```
We don't actually need a `using Test` here. Because of the way these files are included we can assume that `Test` is loaded beforehand.

LilithHafner reviewed (commented):
Lovely tests, thanks!

tsbranquinho committed:
test: remove redundant `using Test` in test/complex.jl

Co-authored-by: Lilith Orion Hafner <lilithhafner@gmail.com>

tsbranquinho commented:
Hey @LilithHafner, I have applied the suggested changes.  
Let me know if there's anything else I should update.  
Thanks for the review, I appreciate the feedback!

LilithHafner reviewed (approved):
Thanks! This looks good to me (LGTM). We'll wait to make sure that CI passes (or at least all the checks that pass on master also pass here) and then merge. Thank you for you contribution!
